src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -660,7 +660,9 @@ export async function activateRoslynLanguageServer(
         const capabilities = await _languageServer.getServerCapabilities();
 
         if (capabilities._vs_onAutoInsertProvider) {
-            if (!capabilities._vs_onAutoInsertProvider._vs_triggerCharacters.includes(change.text)) {
+            const changeTrimmed = change.text.replace(/(?!^)\s+/g, '');
```

Comment:
Why does whitespace in the input text cause indentation to not get properly applied, and why does removing the whitespace fix it?

Reply:
The trigger character is '\n' so it doesn't match otherwise. Also the server side expects the text to match exactly.

Reply:
this needs comments explaining what the heck is going on (including exaplining the regex :)).